File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int n1 = 0; n1 < nums.size(); n1++){
            for (int n2 = n1 + 1; n2 < nums.size(); n2++){
                if (nums[n1] + nums[n2] == target && n1 != n2){
                    return {n1, n2};
                }
            }
        }
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        a = 0
        b = len(nums) - 1

        while a <= b:
            mid = (a + b) // 2

            if nums[mid] == target:
                return mid

            if nums[a] <= nums[mid]:
                if nums[a] <= target < nums[mid]:
                    b = mid - 1
                else:
                    a = mid + 1
            else:
                if nums[mid] < target <= nums[b]:
                    a = mid + 1
                else:
                    b = mid - 1

        return -1